From: David Osborne <david@qcode.co.uk>
Date: Thu, 30 Jul 2015 12:28:12 +0100
Subject: Typo

---
 nslog/nslog.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nslog/nslog.c b/nslog/nslog.c
index 1c480e8..018bdbd 100644
--- a/nslog/nslog.c
+++ b/nslog/nslog.c
@@ -688,7 +688,7 @@ LogTrace(void *arg, Ns_Conn *conn)
     for (h = logPtr->extheaders; *h != NULL; h++) {
         Ns_DStringNAppend(&ds, " \"", 2);
         p = Ns_SetIGet(conn->headers, *h);
-        if (p == NULL) {
+        if (p != NULL) {
             AppendEscaped(&ds, p);
         }
         Ns_DStringNAppend(&ds, "\"", 1);
