From: David Osborne <david@qcode.co.uk>
Date: Thu, 5 Mar 2015 15:46:50 +0000
Subject: Update rpath var names

---
 configure.ac |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 1ffb3b4..cff7d62 100644
--- a/configure.ac
+++ b/configure.ac
@@ -133,8 +133,8 @@ case "$system" in
     LDSO="\$(LDLIB)"
     CCRPATHS="\$(CCRPATH)"
     LDRPATHS="\$(LDRPATH)"
-    CCRFLAG=$TCL_CC_SEARCH_FLAGS
-    LDRFLAG=$TCL_LD_SEARCH_FLAGS
+    CCRFLAG=$CC_SEARCH_FLAGS
+    LDRFLAG=$LD_SEARCH_FLAGS
     if test "$CCRFLAG" = "" ; then
       CCRPATH=
     fi
