From: David Osborne <david@qcode.co.uk>
Date: Tue, 7 Jul 2020 10:50:56 +0100
Subject: ISO_BUILD_DATE to match upstream build date

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 62509aa..48c6bc9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -57,7 +57,7 @@ AC_SUBST([NS_VERSION],     [\"${NS_MAJOR_VERSION}.${NS_MINOR_VERSION}\"])
 AC_SUBST([NS_PATCH_LEVEL], [\"$PACKAGE_VERSION\"])
 #AC_SUBST([NS_PATCH_LEVEL], [\"${NS_MAJOR_VERSION}.${NS_MINOR_VERSION}.${NS_RELEASE_SERIAL}\"])
 
-AC_SUBST([ISO_BUILD_DATE], [`date +%Y-%m-%d`])
+AC_SUBST([ISO_BUILD_DATE], [2020-01-06])
 
 
 
