From: David Osborne <david@qcode.co.uk>
Date: Thu, 14 Jul 2016 12:37:30 +0100
Subject: Add to disclean files

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 90b7eb9..c47f0fc 100644
--- a/Makefile
+++ b/Makefile
@@ -267,8 +267,8 @@ clean-bak: clean
 
 distclean: clean
 	$(RM) config.status config.log config.cache autom4te.cache aclocal.m4 configure \
-	include/{Makefile.global,Makefile.module,config.h,config.h.in,stamp-h1} \
-	naviserver-$(NS_PATCH_LEVEL).tar.gz sample-config.tcl
+	include/{Makefile.global,Makefile.module,config.h,config.h.in,stamp-h1,nsconfig.h,nsversion.h} \
+	naviserver-$(NS_PATCH_LEVEL).tar.gz sample-config.tcl naviserver.rdf version_include.man
 
 dist: clean
 	$(RM) naviserver-$(NS_PATCH_LEVEL)
