From: David Osborne <david@qcode.co.uk>
Date: Wed, 28 Sep 2016 16:05:13 +0100
Subject: Add bash to Makefile so distclean will work

---
 Makefile              |    2 ++
 debian/patches/series |    1 -
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 1275f7c..cba2191 100644
--- a/Makefile
+++ b/Makefile
@@ -29,6 +29,8 @@
 #
 #
 
+SHELL=/bin/bash
+
 NSBUILD=1
 include include/Makefile.global
 
diff --git a/debian/patches/series b/debian/patches/series
index 405929b..6cb8266 100644
--- a/debian/patches/series
+++ b/debian/patches/series
@@ -1,2 +1 @@
-0001-Specify-bash-in-Makefile-so-distclean-will-work.patch
 0002-Add-files-to-distclean.patch
