From: David Osborne <david@qcode.co.uk>
Date: Tue, 5 Apr 2022 16:05:35 +0100
Subject: add stray files to distclean

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 6060321..17ec827 100644
--- a/Makefile
+++ b/Makefile
@@ -302,8 +302,8 @@ clean-bak: clean
 	@find . -name '*~' -exec rm "{}" ";"
 
 distclean: clean
-	$(RM) config.status config.log config.cache autom4te.cache aclocal.m4 configure \
-	include/{Makefile.global,Makefile.module,config.h,config.h.in,stamp-h1} \
+	$(RM) config.status config.log config.cache autom4te.cache aclocal.m4 configure naviserver.rdf version_include.man \
+	include/{Makefile.global,Makefile.module,config.h,config.h.in,stamp-h1,nsconfig.h,nsconfig.h.in,nsversion.h} \
 	naviserver-$(NS_PATCH_LEVEL).tar.gz sample-config.tcl
 
 config.guess:
