From: David Osborne <david@qcode.co.uk>
Date: Wed, 11 Mar 2015 10:27:37 +0000
Subject: Fix var initialisation in Ns_Cork

---
 nsd/sockfile.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/nsd/sockfile.c b/nsd/sockfile.c
index fe78f5b..2b9faf4 100644
--- a/nsd/sockfile.c
+++ b/nsd/sockfile.c
@@ -379,14 +379,14 @@ ssize_t pread(int fd, char *buf, size_t count, off_t offset)
  *      success (0 or 1)
  *
  * Side effects:
- *      Switch TCP send state, pothentially update sockPtr->flags
+ *      Switch TCP send state, potentially update sockPtr->flags
  *
  *----------------------------------------------------------------------
  */
 int
 Ns_SockCork(Ns_Sock *sock, int cork)
 {
-    int success = 1;
+    int success = 0;
 #ifdef TCP_CORK
     Sock *sockPtr = (Sock *)sock;
 
