From: David Osborne <david@qcode.co.uk>
Date: Mon, 3 Aug 2015 15:40:53 +0100
Subject: Null log chars fix

---
 nslog/nslog.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/nslog/nslog.c b/nslog/nslog.c
index 018bdbd..6724aa1 100644
--- a/nslog/nslog.c
+++ b/nslog/nslog.c
@@ -652,7 +652,7 @@ LogTrace(void *arg, Ns_Conn *conn)
         if (p != NULL) {
             AppendEscaped(&ds, p);
         }
-        Ns_DStringNAppend(&ds, "\"", 2);
+        Ns_DStringNAppend(&ds, "\"", 1);
     }
 
     /*
@@ -678,7 +678,7 @@ LogTrace(void *arg, Ns_Conn *conn)
         Ns_DStringPrintf(&ds, " %" PRIu64 ".%06ld", (int64_t)queueTime.sec,     queueTime.usec);
         Ns_DStringPrintf(&ds, " %" PRIu64 ".%06ld", (int64_t)filterTime.sec,    filterTime.usec);
         Ns_DStringPrintf(&ds, " %" PRIu64 ".%06ld", (int64_t)runTime.sec,       runTime.usec);
-        Ns_DStringNAppend(&ds, "\"", 2);
+        Ns_DStringNAppend(&ds, "\"", 1);
     }
 
     /*
