From: David Osborne <david@qcode.co.uk>
Date: Fri, 30 Sep 2016 10:07:24 +0100
Subject: bug fix for mime-type in fastpath

---
 nsd/fastpath.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/nsd/fastpath.c b/nsd/fastpath.c
index 3319977..4ed4111 100644
--- a/nsd/fastpath.c
+++ b/nsd/fastpath.c
@@ -509,6 +509,14 @@ FastReturn(Ns_Conn *conn, int statusCode, const char *mimeType, const char *file
         return status;
     }
 
+    /*
+     * Determine the mime type if not given based on the requested
+     * file name (without a potential gz suffix).
+     */
+    if (mimeType == NULL) {
+        mimeType = Ns_GetMimeType(file);
+    }
+
     Tcl_DStringInit(dsPtr);
 
     /*
@@ -554,14 +562,6 @@ FastReturn(Ns_Conn *conn, int statusCode, const char *mimeType, const char *file
 	    }
 	}
     }
-
-    /*
-     * Determine the mime type if not given.
-     */
-
-    if (mimeType == NULL) {
-        mimeType = Ns_GetMimeType(file);
-    }
     
     /*
      * For no output (i.e., HEAD request), just send required
