From: David Osborne <david@qcode.co.uk>
Date: Thu, 28 Jul 2016 10:03:17 +0100
Subject: Missing closing brace in commandlist.man

---
 doc/src/naviserver/commandlist.man |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doc/src/naviserver/commandlist.man b/doc/src/naviserver/commandlist.man
index 1674c41..cad0edd 100644
--- a/doc/src/naviserver/commandlist.man
+++ b/doc/src/naviserver/commandlist.man
@@ -417,7 +417,7 @@ List of all currently available NaviServer commands.
 [call [cmd ns_proxy_handler_http]]
 [call [cmd ns_shortcut_filter]]
 [call [cmd ns_fileread]]
-[call [cmd ns_sourceproc]
+[call [cmd ns_sourceproc]]
 [call [cmd ns_upload_stats]]
 [call [cmd ns_writer]]
 [call [cmd ns_moduleload]]
