From: Qcode Software Ltd <hackers@qcode.co.uk>
Date: Mon, 10 Apr 2017 11:01:49 +0100
Subject: nsssl shutdown error handling

---
 nsssl/nsssl.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/nsssl/nsssl.c b/nsssl/nsssl.c
index b0e7861..58014da 100644
--- a/nsssl/nsssl.c
+++ b/nsssl/nsssl.c
@@ -679,6 +679,13 @@ Close(Ns_Sock *sock)
             shutdown(SSL_get_fd(sslPtr->ssl), SHUT_RDWR);
             r = SSL_shutdown(sslPtr->ssl);
         }
+        if (r == -1) {
+            unsigned long err = ERR_get_error();
+            
+            if (err != 0) {
+                Ns_Log(Notice, "SSL_shutdown has failed: %s", ERR_error_string(err, NULL));
+            }
+        }
         SSL_free(sslPtr->ssl);
         ns_free(sslPtr);
     }
