From: David Osborne <david@qcode.co.uk>
Date: Thu, 16 Feb 2017 09:56:00 +0000
Subject: allow headerless replies

---
 nsd/driver.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/nsd/driver.c b/nsd/driver.c
index 709c05f..32f3d0b 100644
--- a/nsd/driver.c
+++ b/nsd/driver.c
@@ -4835,9 +4835,10 @@ NsWriterQueue(Ns_Conn *conn, size_t nsend, Tcl_Channel chan, FILE *fp, int fd,
         Ns_Log(DriverDebug, "add header (fd %d)\n", fd);
         conn->flags |= NS_CONN_SENTHDRS;
         (void)Ns_CompleteHeaders(conn, nsend, 0u, &ds);
-
-        wrSockPtr->headerString = ns_strdup(Tcl_DStringValue(&ds));
         headerSize = (size_t)Ns_DStringLength(&ds);
+        if (headerSize > 0u) {
+            wrSockPtr->headerString = ns_strdup(Tcl_DStringValue(&ds));
+        }
         Ns_DStringFree(&ds);
     } else {
         headerSize = 0u;
