From: Qcode Software Ltd <hackers@qcode.co.uk>
Date: Tue, 3 Oct 2017 14:13:38 +0100
Subject: checkforproxy logix fix

---
 debian/changelog | 6 ++++++
 nsd/queue.c      | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/debian/changelog b/debian/changelog
index afdb467..0af8683 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,9 @@
+naviserver (4.99.16~20171002-1+deb8u2) jessie; urgency=medium
+
+  * Patch to fix logic in checkforproxy check
+
+ -- Qcode Software Ltd <hackers@qcode.co.uk>  Tue, 03 Oct 2017 14:11:20 +0100
+
 naviserver (4.99.16~20171002-1+deb8u1) jessie; urgency=medium
 
   * Import from upstream (includes checkforproxy ns_server option)
diff --git a/nsd/queue.c b/nsd/queue.c
index ace1241..e770cda 100644
--- a/nsd/queue.c
+++ b/nsd/queue.c
@@ -2391,7 +2391,7 @@ AppendConn(Tcl_DString *dsPtr, const Conn *connPtr, const char *state, bool chec
 
             if ( checkforproxy ) {
                 p = Ns_SetIGet(connPtr->headers, "X-Forwarded-For");
-                if (p == NULL || (*p != '\0') || strcasecmp(p, "unknown") == 0) {
+                if (p == NULL || (*p == '\0') || strcasecmp(p, "unknown") == 0) {
                     /*
                      * Lookup of header field failed, use upstream peer
                      * address.
