From: Qcode Software Ltd <hackers@qcode.co.uk>
Date: Wed, 22 Feb 2017 17:00:14 +0000
Subject: choose driver named in ns_connchan

---
 nsd/driver.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/nsd/driver.c b/nsd/driver.c
index 384c93f..8901272 100644
--- a/nsd/driver.c
+++ b/nsd/driver.c
@@ -5924,15 +5924,19 @@ NSDriverClientOpen(Tcl_Interp *interp, const char *driverName,
             Ns_Log(DriverDebug, "... check Driver proto <%s> server %s name %s location %s",
                    drvPtr->protocol, drvPtr->server, drvPtr->threadName, drvPtr->location);
             if (STREQ(drvPtr->protocol, protocol)) {
-                if (driverName != NULL && STREQ(drvPtr->moduleName, driverName)) {
+                if (driverName == NULL) {
                     break;
-                } else {
+                } else if (STREQ(drvPtr->moduleName, driverName)) {
                     break;
                 }
             }
         }
         if (drvPtr == NULL) {
-            Ns_TclPrintfResult(interp, "no driver for protocol '%s' found", protocol);
+            if (driverName != NULL) {
+                Ns_TclPrintfResult(interp, "no driver for protocol '%s' & driver name '%s' found", protocol, driverName);
+            } else {
+                Ns_TclPrintfResult(interp, "no driver for protocol '%s' found", protocol);
+            }
             result = TCL_ERROR;
 
         } else if (portString != NULL) {
