From: David Osborne <david@qcode.co.uk>
Date: Fri, 10 Jun 2022 15:05:20 +0100
Subject: fix ns_writer test typos

---
 tests/ns_writer.test | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/ns_writer.test b/tests/ns_writer.test
index 1512655..d49be99 100644
--- a/tests/ns_writer.test
+++ b/tests/ns_writer.test
@@ -20,14 +20,14 @@ test ns_writer-1.2 {basic syntax: wrong argument} -body {
 test ns_writer-1.3 {basic operation} -body {
     catch {ns_writer list -foo} errorMsg
     set _ $errorMsg
-}-cleanup {
+} -cleanup {
     unset -nocomplain _
 } -match exact -result {wrong # args: should be "ns_writer list ?-server server?"}
 
 test ns_writer-1.4 {basic operation} -body {
     catch {ns_writer list -server test foo} errorMsg
     set _ $errorMsg
-}-cleanup {
+} -cleanup {
     unset -nocomplain _
 } -match exact -result {wrong # args: should be "ns_writer list ?-server server?"}
 
