From: David Osborne <david@qcode.co.uk>
Date: Tue, 7 Jul 2020 14:35:07 +0100
Subject: cast to avoid compiler warning

---
 nslog/nslog.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nslog/nslog.c b/nslog/nslog.c
index 1428e0f..156b2fc 100644
--- a/nslog/nslog.c
+++ b/nslog/nslog.c
@@ -347,7 +347,7 @@ ParseExtendedHeaders(Log *logPtr, const char *str)
         int    argc;
         char **argv;
 
-        if (Tcl_SplitList(NULL, str, &argc, &argv) != TCL_OK) {
+        if (Tcl_SplitList(NULL, str, &argc, (const char***)&argv) != TCL_OK) {
             Ns_Log(Error, "nslog: invalid 'extendedHeaders' parameter: '%s'", str);
             result = NS_ERROR;
 
