From: Qcode Software Ltd <hackers@qcode.co.uk>
Date: Mon, 28 Jan 2019 11:03:48 +0000
Subject: exclude tcl namespaces

---
 tcl/nstrace.tcl | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/tcl/nstrace.tcl b/tcl/nstrace.tcl
index 9bfbc41..8e12c4a 100644
--- a/tcl/nstrace.tcl
+++ b/tcl/nstrace.tcl
@@ -733,7 +733,10 @@ ns_runonce {
             #
             foreach pn [info procs ${nsp}::*] {
                 set orig [namespace origin $pn]
-                if {$orig ne [namespace which -command $pn]} {
+                if {
+		   $orig ne [namespace which -command $pn]
+		   && ![string match "::tcl::*" $orig
+		} {
                     append import "namespace import -force [list $orig]" \n
                 } else {
                     append script [_procscript $pn]
